### PR TITLE
abstracted palettes, colors, and palette manager

### PR DESCRIPTION
**Pull request by gravestench**

### CODE REVIEW

File: d2common/d2fileformats/d2dat/dat_color.go [hunk 1]
```diff
@@ -2,7 +2,59 @@ package d2dat
...
+	rgba := uint32(c.r)<<bitShiftR
+	rgba += uint32(c.g)<<bitShiftG
+	rgba += uint32(c.b)<<bitShiftB
+	rgba += uint32(c.b)<<bitShiftA
+
+	return rgba
+}
+
+// SetRGBA sets the color components using the given RGBA form
+func (c *DATColor) SetRGBA(rgba uint32) {
```
Comment (essial) [inline d2common/d2fileformats/d2dat/dat_color.go:54]: Can we get a Get/Set BGRA as well?

Comment (gravestench) [reply d2common/d2fileformats/d2dat/dat_color.go:54]: sure, i'll add it in a bit

Comment (gravestench) [reply d2common/d2fileformats/d2dat/dat_color.go:49]: :)

**Comment by gravestench** [comment]
this was merged with #544

**Pull request closed without merging**